### PR TITLE
build(deps): bump go-simpler.org/sloglint from 0.5.1 to 0.6.0

### PR DESCRIPTION
**Pull request by tmzane**
In continuation to #4512

Not sure about the new name, `context` does not seem descriptive to me. Maybe something like `no-contextless`? Weird, but easier to understand.

Please don't merge it yet, I'm going to update the pinned version later.

I've also improved the report message slightly.

https://github.com/go-simpler/sloglint/compare/v0.5.1...v0.6.0

### CODE REVIEW

File: pkg/golinters/sloglint/sloglint.go [hunk 1]
```diff
@@ -14,14 +14,17 @@ func New(settings *config.SlogLintSettings) *goanalysis.Linter {
...
-			ContextOnly:    settings.ContextOnly,
+			NoGlobal:       settings.NoGlobal,
+			ContextOnly:    settings.Context,
 			StaticMsg:      settings.StaticMsg,
 			NoRawKeys:      settings.NoRawKeys,
 			KeyNamingCase:  settings.KeyNamingCase,
 			ArgsOnSepLines: settings.ArgsOnSepLines,
 		}
+		if settings.ContextOnly && settings.Context == "" {
+			opts.ContextOnly = "all"
```
Comment (tmzane) [inline pkg/golinters/sloglint/sloglint.go:26]: Should we print a deprecation notice here?

Comment (ldez) [reply pkg/golinters/sloglint/sloglint.go:26]: the deprecated and the fallback you should be handled inside the `Loader`
https://github.com/golangci/golangci-lint/blob/44c070af0829b1f87ecf754e2cccc4e74a22b27f/pkg/config/loader.go#L358

**Review by ldez — CHANGES_REQUESTED** [review]
You should not modify:
- `.golangci.reference.yml`
- `jsonschema/golangci.jsonschema.json`

Also `context-only` should be removed from:
- `.golangci.next.reference.yml`
- `jsonschema/golangci.next.jsonschema.json`

I think `context` is good, at least less weird than `no-contextless` (IMHO it's not easier to understand)
The values can be improved, for example: `all` -> `always`, `scope` -> `existing`

**Comment by tmzane** [comment]
> Also `context-only` should be removed from:
> 
>     * `.golangci.next.reference.yml`
> 
>     * `jsonschema/golangci.next.jsonschema.json`

If it is removed from the next version schema, won't it become illegal to use in the config? Thus, breaking backward compatibility with existing configs.

**Comment by ldez** [comment]
> won't it become illegal to use in the config? 

"illegal" no :smile:

The JSON schema validation is only called explicitly, but the deprecation warning will always display, so everything will still work as expected.

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**